wger/core/static/js/wger-core.js

```diff
@@ -459,6 +459,87 @@ function updateAllExerciseFormset() {
   }
 }
 
+function  setDropsetScale() {
+  var setValue = $('#drop-set-set-value').val();
+
+  $('#id_sets').attr('value', setValue);
+  updateAllExerciseFormset();
+}
+
+function populateDropSetField() {
+  var exerciseId = $('.ajax-exercise-select a').parent('div').find('input').val();
+  var numOfSets = $('#id_sets').val();
+  var maxWeight = $('#dropset-max-weight').val();
+  var minReps = $('#dropset-min-rep').val();
+  var i;
+  var weightPercent = 1;
+  // var percentReduction = 0.2;
+  var additionalReps = 0;
+  var weightValue;
+  var repsValue;
+
+
+  for (i = 0; i < parseInt(numOfSets, 10); i++) {
+    if (i === 0) {
+      // no-op
+    } else {
+      weightPercent -= (parseInt($('#dropset-weight-drop').val(), 10) / 100);
+      additionalReps += parseInt($('#reps-increment-value').val(), 10);
+    }
+
+    if (parseInt(weightPercent * maxWeight, 10) > 0) {
+      weightValue = parseInt(weightPercent * maxWeight, 10);
+    } else {
+      weightValue = 0;
+    }
+
+    if (parseInt(additionalReps + parseInt(minReps, 10), 10) > 0) {
+      repsValue = parseInt(additionalReps +
+      parseInt(minReps, 10), 10);
+    } else {
+      repsValue = 0;
+    }
+
+    $('#id_exercise' + exerciseId + '-' + i + '-weight').attr('value', weightValue);
+    $('#id_exercise' + exerciseId + '-' + i + '-reps').attr('value', repsValue);
+  }
+}
+
+function setDropSet() {
+  var setScale = $('#id_sets');
+  var typeValue = $('#drop-set');
+  var maxWeight = $('#max-weight');
+  var minReps = $('#min-reps');
+  var helpLink = $('#dropset-help');
+  var dropSet = $('#id_dropset');
+  var weightIndex = $('#weight-drop');
+  var repsIndex = $('#reps-increment');
+
+  if (typeValue.val() === 'drop-set') {
+    dropSet.attr('checked', 'checked');
+    setScale.attr('min', '3');
+    setScale.attr('max', '5');
+    setScale.attr('value', '4');
+    $('#id_sets_value').html('4');
+    maxWeight.removeClass('hidden');
```

Comment:
Make a function that has the identical lines and invoke the function here instead.